aws/resource_aws_wafv2_regex_pattern_set.go

```diff
@@ -0,0 +1,267 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/wafv2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsWafv2RegexPatternSet() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsWafv2RegexPatternSetCreate,
+		Read:   resourceAwsWafv2RegexPatternSetRead,
+		Update: resourceAwsWafv2RegexPatternSetUpdate,
+		Delete: resourceAwsWafv2RegexPatternSetDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				idParts := strings.Split(d.Id(), "/")
+				if len(idParts) != 3 || idParts[0] == "" || idParts[1] == "" || idParts[2] == "" {
+					return nil, fmt.Errorf("Unexpected format of ID (%q), expected ID/NAME/SCOPE", d.Id())
+				}
+				id := idParts[0]
+				name := idParts[1]
+				scope := idParts[2]
+				d.SetId(id)
+				d.Set("name", name)
+				d.Set("scope", scope)
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(1, 256),
+			},
+			"lock_token": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 128),
+			},
+			"regular_expression_list": {
```

Comment:
how would we feel about changing this attribute to just a `Set` of `Elem: &schema.Schema{Type: schema.TypeString}` (i'm thinking a la `wafregional_regex_pattern_set` schema)? since the API expects a list of strings in the request, perhaps it could help to simplify things here and not require more complex flattening methods.